src/java.base/share/classes/jdk/internal/classfile/impl/ClassFileImpl.java

```diff
@@ -132,7 +132,11 @@ public void accept(ClassBuilder builder) {
 
     @Override
     public List<VerifyError> verify(ClassModel model) {
-        return VerifierImpl.verify(model, classHierarchyResolverOption().classHierarchyResolver(), null);
+        try {
+            return VerifierImpl.verify(model, classHierarchyResolverOption().classHierarchyResolver(), null);
+        } catch (IllegalArgumentException verifierInitializationError) {
+            return List.of(new VerifyError(verifierInitializationError.getMessage()));
```

Comment:
Note that the list returned by `VerifierImpl.verify(…)` is mutable, whereas `List.of(…)` is unmodifiable.

Reply:
Good observation; we should check that all collection-returning APIs in ClassFile API are immutable.